### ISSUE TITLE
[TimePicker] Design review - Contrast of focus visible is too low

### ISSUE THREAD

**Issue by noraleonte** [mui/mui-x#14742]
The focus visible has barely any contrast and should be adjusted

https://github.com/mui/mui-x/assets/3165635/a05c1593-d70f-4357-a8df-63d7ce7beb59
